charts/datafold/charts/dfshell/templates/serviceaccount.yaml

```diff
@@ -2,12 +2,15 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ include "dfshell.serviceAccountName" . }}
+  name: {{ .Values.serviceAccount.name }}
   labels:
     {{- include "dfshell.labels" . | nindent 4 }}
-  {{- with .Values.serviceAccount.annotations }}
   annotations:
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+    {{- if .Values.serviceAccount.annotations }}
+    {{- toYaml .Values.serviceAccount.annotations | nindent 4 }}
+    {{- end }}
+    {{- if .Values.serviceAccount.roleArn }}
```

Comment:
It may be better to also filter on cloud type (aws). That way we don't set this EKS role for GCP / azure where these helm-charts are also deployed. It won't crash, but it would have weird annotations in there.